src/components/BlockchainInfo.js

```diff
@@ -592,37 +598,77 @@ class BlockchainInfo extends React.Component {
     }
   }
 
-  async fundAsset(assetId, amount, onSuccessConfirmationPopup, onFailureContributionPopup, amountDollars) {
+  buildAndUpdateNotification(notificationId, type, status, params){
+    this.updateNotification(notificationId, {
+      [type]: {
+        ...params,
+      },
+      status,
```

Comment:
Future you will be like wat haha, jk, I think this is fine.

Reply:
present me is already looking at the current code and having that reaction. I'm prepared for anything!

Go is getting there though

Reply:
🎉